### PR TITLE
feat(amf): Proto fields to support stateless feature

### LINKED ISSUE #10883: Add necessary proto fields to support Stateless Feature.

### PR DESCRIPTION
**Pull request by shashidhar-patil**
Signed-off-by: shashidhar-patil <shashidhar.patil@wavelabs.ai>

### Proto Changes added to support stateless feature.

**Implementation:**
1. For **ue_m5gmm_context_s reusing existing UeContext proto**. Added required placeholders to the existing UeContext proto to support 5G specific fields.
proto_UeContext

2. For **amf_context_s reusing existing EmmContext proto**. Added required placeholders to the existing EmmContext proto to support 5G specific fields.
proto_AmfContext

3. For **amf_security_context_s reusing existing EmmSecurityContext proto**.Added required placeholders to the existing EmmSecurityContext proto to support 5G specific fields.
proto_AmfSecurityContext

4. For **smf_context_s added new proto message SmfContext**.
proto_SmfContext

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/protos/oai/nas_state.proto [hunk 1]
```diff
@@ -475,4 +479,68 @@ message EmmContext {
   NewAttachInfo new_attach_info = 67;
   UeAdditionalSecurityCapability ue_additional_security_capability = 68;
   // TODO: add remaining emm_context elements
+
+  // 5G Guti
+  Guti_m5 m5_guti = 91;
+  Guti_m5 m5_old_guti = 92;
+  // 5G proto
+  map<uint32,SmfContext> smf_context_m =93;
```
Comment (ssanadhya) [inline lte/protos/oai/nas_state.proto]: What is the key for this hashmap? Please use a field name which includes what the key is.

File: lte/protos/oai/nas_state.proto [hunk 2]
```diff
@@ -475,4 +479,68 @@ message EmmContext {
...
+    uint32 m_tmsi = 5;
+  }
+
+  message Smf_Proc_Data {
+    uint32 pdu_session_id = 1;
+    uint32 pti = 2;
+    uint32 message_type = 3;
+    uint32 max_uplink = 4;
+    uint32 max_downlink = 5;
+    uint32 pdu_session_type = 6;
```
Comment (ssanadhya) [inline lte/protos/oai/nas_state.proto]: Enum as used in session_manager.proto

**Review by ssanadhya — CHANGES_REQUESTED** [review]

File: lte/protos/oai/mme_nas_state.proto [hunk 3]
```diff
@@ -150,6 +150,10 @@ message UeContext {
   google.protobuf.Timestamp time_paging_response_timer_started = 209;
   uint32 paging_retx_count = 210;
   google.protobuf.Timestamp time_ics_rsp_timer_started = 211;
+
+  // 5G protos
+  uint32 amf_ue_ngap_id = 221;
+  uint32 ue_context_request = 222;
```
Comment (ardzoht) [inline lte/protos/oai/mme_nas_state.proto:158]: can we rename this to be more explicit on what this field represents? if there is no such name, add a comment on top of the field

File: lte/protos/oai/nas_state.proto (hunk 1 above)
Comment (ardzoht) [inline lte/protos/oai/nas_state.proto]:
```suggestion
  map<uint32,SmfContext> smf_context_map = 93;
```

Comment (ardzoht) [reply lte/protos/oai/nas_state.proto]: +1

**Review by ardzoht — CHANGES_REQUESTED** [review]

File: lte/protos/oai/nas_state.proto (hunk 2 above)
Comment (shashidhar-patil) [reply lte/protos/oai/nas_state.proto]: For this change, the application code change requested in #10853 is needed. The `enum PduSessionType` should be redefined in nas_state.proto similar to how it is defined in session_manager.proto, right?

File: lte/protos/oai/mme_nas_state.proto (hunk 3 above)
Comment (shashidhar-patil) [reply lte/protos/oai/mme_nas_state.proto:158]: Added the comment.

File: lte/protos/oai/nas_state.proto (hunk 1 above)
Comment (shashidhar-patil) [reply lte/protos/oai/nas_state.proto]: Addressed this review comment.

Comment (shashidhar-patil) [reply lte/protos/oai/nas_state.proto]: Review comment addressed.

File: lte/protos/oai/nas_state.proto (hunk 2 above)
Comment (ssanadhya) [reply lte/protos/oai/nas_state.proto]: The changes here can happen in parallel with the changes in #10853 , and the PR with the state converter logic can  stitch them together.

File: lte/protos/oai/nas_state.proto [hunk 4]
```diff
@@ -475,4 +479,68 @@ message EmmContext {
...
+
+  // 5G Guti
+  Guti_m5 m5_guti = 91;
+  Guti_m5 m5_old_guti = 92;
+  // 5G proto
+  map<uint32,SmfContext> pdu_session_id_smf_context_map =93;
+
+  }
+
+  // guti_m5_t
```
Comment (ssanadhya) [inline lte/protos/oai/nas_state.proto]: Please move all the newly defined messages before EmmContext message definition. Definition should come before usage.

**Review by ssanadhya — CHANGES_REQUESTED** [review]

File: lte/protos/oai/nas_state.proto (hunk 4 above)
Comment (ardzoht) [reply lte/protos/oai/nas_state.proto]: +1 on this

**Review by ardzoht — CHANGES_REQUESTED** [review]
Requesting changes for all new messages that are part of `EmmContext` should be defined before

File: lte/protos/oai/nas_state.proto (hunk 2 above)
Comment (shashidhar-patil) [reply lte/protos/oai/nas_state.proto]: Okay sure, made the change.

File: lte/protos/oai/nas_state.proto (hunk 4 above)
Comment (shashidhar-patil) [reply lte/protos/oai/nas_state.proto]: Addressed this review comment

**Review by ssanadhya — APPROVED** [review]
LGTM
Can be merged once @ardzoht also approves.

**Review by ardzoht — APPROVED** [review]
lgtm

**Pull request merged**